backend/spec/performance/invoice_list_query_performance_spec.rb

```
@@ -0,0 +1,170 @@
+    it 'performs significantly faster with the composite index' do
+      query_sql = build_invoice_list_sql(company.id)
+      
+      # Measure query execution time
+      execution_time = measure_query_time(query_sql)
+      
+      # With 500 invoices and proper indexing, query should be very fast
+      # This threshold should be adjusted based on your performance requirements
+      expect(execution_time).to be < 50.0 # 50ms threshold
+    end
+
+    it 'maintains consistent performance regardless of company size' do
```

RuboCop (1.75.5)

[convention] Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

(Style/StringLiterals)